### DISCUSSION TITLE
file locking when writing data

### DISCUSSION THREAD

**Discussion by aodhanmci** [epochpic/epoch#237] [Q&A]
Hello, 
is it possible to compile EPOCH (the SDF module in particular) that allows for file-locking? As I understand it, when writing the SDF files, file-locking is required to be switched off. But it is on by default for our cluster since it is apparently much higher performance (so they won't switch it off).
Sorry if this is a bit naive, I don't understand this requirement very well.

Aodhan

**Comment by keithbennett** [comment]
EPOCH just uses standard MPI-IO and has no specific requirement for file locking. What issues are you experiencing?

**Reply by aodhanmci** [reply]
The job will crash when trying to write some SDF files - usually a UCX error. The management initially pointed to file locking being the problem so when they mounted a small amount of storage for us without file locking there were no problems. I will run a few jobs now and see if it crashes out and I will show the error

**Reply by keithbennett** [reply]
What MPI libraries are being used and what is the underlying MPI-IO implementation? For a long while, all MPI-IO implementations were based on ROMIO but OpenMPI have an alternative library of their own now. Locking behaviour can often be controlled using environment variables or hints passed via the MPI_Info object. I think a bit of experimentation is required. I found a discussion on the mpich forums which might help: https://lists.mpich.org/pipermail/discuss/2016-November/000308.html
